### PR TITLE
feat: parse profile name from dbt_project

### PR DESCRIPTION
**Pull request by bastienboutonnet**
# Description
In order for `dbt-sugar` to know which profile to choose in order to connect to a database it needs to read the profile name from dbt's `dbt_project.yml`.

- The `sugar_config.yml` contains a list (currently we only support one entry) of dbt projects under scope. This list is made of dicts which contain `name` and `path` of dbt projects.
- The `DbtProject` object then loads the `dbt_project.yml` file from it's provided path and stores the `profile_name` which will be provided to `DbtProfile`.

This PR also implements reading the correct `target` profile from the `profiles.yml` by either getting it from `--target` on the CLI or from the `target:` yaml field in `profiles.yml`. Priority is given to the CLI argument as it is "closer" to the user.

# How was the change tested
I modified the unit tests to work and it seems good locally but I will wait to merge this branch until #51 is merged so that I can test this fully and with more corner cases.

# Issue Information
Partially addresses #38 because it introduces a necessary regression which renders `dbt-sugar` unable to have more than one dbt project in scope until we refactor how we store models associated to projects which will enable us to dynamically feed the right profile and credentials to the `DocumentationTask` db query step.

# Checklist

(Ideally, all boxes are checked by the time we merged the PR, if you don't know how to do any of these don't hesitate to say so in the PR and we'll help you out.)

- [x] I formatted my PR name according to [CONTRIBUTING.md](CONTRIBUTING.md)
- [ ] I added a news fragment to help populating the changelog as encouraged in [CONTRIBUTING.md](CONTRIBUTING.md)
- [x] I added "Closes #<issue_number>" in the "Issue Information" section (if no issue, feel free to tick thick the box anyway).

### CODE REVIEW

File: dbt_sugar/core/clients/dbt.py [hunk 1]
```diff
@@ -71,22 +143,29 @@ def profiles_dir(self):
...
+            return profile_dict["outputs"].get(self._target_name)
         else:
-            raise DbtProfileFileMissing(
-                f"Could not locate `profiles.yml` in {self.profiles_dir.resolve()}."
-            )
+            self._target_name = profile_dict.get("target", str())
+            if self._target_name:
+                return profile_dict["outputs"].get(self._target_name)
+            else:
+                raise TargetNameNotProvided(
```
Comment (virvirlopez) [inline dbt_sugar/core/clients/dbt.py:154]: Really nice messages!

File: tests/test_yaml_helpers.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def test_save_yaml(content, result):
     try:
         save_yaml(temp_file.name, content)
         yaml_content = open_yaml(Path(temp_file.name))
-        assert yaml_content == yaml.load(result)
+        assert yaml_content == yaml.safe_load(result)
```
Comment (virvirlopez) [inline tests/test_yaml_helpers.py:47]: maybe we want to change this, now that we have changed the way of loading the YAML.

**Review by virvirlopez — APPROVED** [review]
Only a little comment. LGTM

File: tests/test_yaml_helpers.py (hunk 2 above)
Comment (bastienboutonnet) [reply tests/test_yaml_helpers.py:47]: Eh I think in the test it's ok no?
